pkg/api/v1beta3/conversion.go

```diff
@@ -28,6 +28,7 @@ func init() {
 		func(label, value string) (string, string, error) {
 			switch label {
 			case "metadata.name",
+				"metadata.namespace",
```

Comment:
I'm resigned to the fact that starting with this syntax is the most expedient thing to do. We can reevaluate later.

In the long run, we should strive to make all field references the same:
* downward API substitution
* client-side cross-object field references
* kubectl -o template
* field selectors
* ObjectReference
* field projection (only return a subset of object fields)